nixos/modules/programs/singularity.nix

```diff
@@ -3,32 +3,83 @@
 with lib;
 let
   cfg = config.programs.singularity;
-  singularity = pkgs.singularity.overrideAttrs (attrs : {
-    installPhase = attrs.installPhase + ''
-      mv $out/libexec/singularity/bin/starter-suid $out/libexec/singularity/bin/starter-suid.orig
-      ln -s /run/wrappers/bin/singularity-suid $out/libexec/singularity/bin/starter-suid
-    '';
-  });
-in {
+in
+{
   options.programs.singularity = {
-    enable = mkEnableOption "Singularity";
+    enable = mkEnableOption "singularity" // {
+      description = ''
+        Whether to install Singularity/Apptainer
+        and add SUID support.
+      '';
+    };
+    package = mkOption {
+      default = pkgs.singularity;
+      defaultText = "pkgs.singularity";
+      description = ''
+        Singularity package to override and install. Typical values are
+        pkgs.singularity, pkgs.singularity-legacy, pkgs.apptainer, and pkgs.singularity-ce.
+      '';
+      type = types.package;
+    };
+    packageOverriden = mkOption {
```

Comment:
I am not sure why we need this option at all. The wrapper should always be added if enableSuid is true. Adding an extra level of indirection makes things just more complicated.

Reply:
Would it be more proper to keep this option but hide it away from users (the manual)? Otherwise, there would be no way to access the overridden package.

Reply:
Just added an example use case of this option in the manual.

Combining with `nixpkgs.overlays`, it enables a user to make all the packages that depends on `singularity` or `apptainer` to use the overridden package.